### DISCUSSION TITLE
Rendering components from when "public path" in webpack is set to different location

### DISCUSSION THREAD

**Discussion by zablik** [inertiajs/inertia#504] [Help (Vue)] [answered]
Hello everyone! I'm not sure if this is an Inertia or Vue issue, but see what happens:

I've changed a public path in my `webpack.mix.js`
```
mix
    .setPublicPath("/public/core")
```
in `app.blade.php` I have also changed the manifest directory
```
<script src="{{ mix('/js/app.js', '/core') }}" defer></script>
```

So, my compiled file structure now is the following:
```
public
--core
------js
--------app.js
--------resources_js_Pages_MyPage_vue.js
--------resources_js_Pages_MyAnotherPage_vue.js
```

When I load my page, `app.js` as well with the styles are loaded perfectly, but when it comes to loading Vue components an application does not know that my public path has been moved and tries to load it from the default `/public/js/resources_js_Pages_MyPage_vue.js` location, no `/core` subfolder prefix is used

What can I do?

Thanks

```
# composer.json:
        "inertiajs/inertia-laravel": "^0.3.5",

# package.json:
        "@inertiajs/inertia": "^0.8.2",
        "@inertiajs/inertia-vue3": "^0.3.4",
        "vue": "^3.0.5",
```

**Comment by zablik** [comment, marked as answer]
Well, I really hate to provide a solution without having a clear understanding WHY does it work exactly like this, but... here it is

in `webpack.mix.js`
```
mix
    .setPublicPath("public/core") // <----
    .setResourceRoot("/core") // <----
    .sass("resources/scss/style.scss", "/css") // <--- removed "/public" prefix
    .js("resources/js/app.js", "/js")  // <--- removed "/public" prefix

..............................

mix.webpackConfig({
    output: {
        publicPath: "/core/" // <----
    },
});
```

I tried to remove any of these fixes and it broke the app. 
btw, without `mix.setResourceRoot("/core")` scripts and styles were working, but images weren't.

I wish someone explained the given behavior. Duplicating a "core" prefix does not seem to be a good practice.

**Reply by claudiodekker** [reply]
Yeah, unfortunately I also have no clue what causes this. The only thing I can tell you 10000% for sure is that this isn't an Inertia-related issue, but rather something caused in/between Webpack and Laravel Mix. If it were Inertia-related, webpack wouldn't even be able to compile your app.

Perhaps it would make sense to ask your question over on the Laravel Mix issue tracker or on the Laracasts forums instead?
